### ISSUE TITLE
Craco with jest 27 createTransformer and _cracoConfig undefined problem after updating from v4

### ISSUE THREAD

**Issue by klougod** [dilanx/craco#425]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@craco/craco@6.2.0` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@craco/craco/lib/features/jest/create-jest-babel-transform.js b/node_modules/@craco/craco/lib/features/jest/create-jest-babel-transform.js
index f4742b9..f732f43 100644
--- a/node_modules/@craco/craco/lib/features/jest/create-jest-babel-transform.js
+++ b/node_modules/@craco/craco/lib/features/jest/create-jest-babel-transform.js
@@ -53,7 +53,7 @@ function createJestBabelTransform(cracoConfig) {
             }
         }
     }
-    return babelJest.createTransformer(craBabelTransformer);
+    return babelJest.default.createTransformer(craBabelTransformer);
 }
 
 module.exports = {
diff --git a/node_modules/@craco/craco/lib/features/jest/jest-babel-transform.js b/node_modules/@craco/craco/lib/features/jest/jest-babel-transform.js
index 13adf42..7fbd9e9 100644
--- a/node_modules/@craco/craco/lib/features/jest/jest-babel-transform.js
+++ b/node_modules/@craco/craco/lib/features/jest/jest-babel-transform.js
@@ -9,7 +9,7 @@ module.exports = {
     ...createJestBabelTransform(),
     process(src, filename, config, transformOptions) {
         if (!jestBabelTransform) {
-            jestBabelTransform = createJestBabelTransform(config.globals._cracoConfig);
+            jestBabelTransform = createJestBabelTransform(config._cracoConfig);
         }
 
         return jestBabelTransform.process(src, filename, config, transformOptions);
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by dilanx** [comment]
@klougod could you share more information about the issue you're having?

**Comment by klougod** [comment]
> @klougod could you share more information about the issue you're having?

Sure, 

I have a CRA ejected application, and after updating from craco v4 to craco v6 and the dependency of babel-jest being the version 27 broke all my tests. Version 24 to 26 wasn't working too but I thought it was due to the update, after updating to babel-jest 27 I got this errors:

Cannot read property createTransformer of undefined
and
Cannot read property _cracoConfig of undefined

An important detail is that everything else works, builds, development environment, etc. It was a problem only in tests.
I can send a PR to fix the issue since I fixed it in my node_modules.

Best regards

**Comment by dilanx** [comment]
Thanks @klougod. Feel free to submit a PR with your fix.

**Comment by klougod** [comment]
> Thanks @klougod. Feel free to submit a PR with your fix and I'll take a look.
Done!

https://github.com/gsoft-inc/craco/pull/432

**Comment by ryuash01** [comment]
Hey first time commenting but first, thank you for this project :D

Don't know if this is related but I was on `@craco/craco": "^6.4.4` and actually had to downgrade to `@craco/craco": "^6.4.3` because of these changes. Currently using `Create React App (CRA) 4.*` without ejecting

**Comment by dilanx** [comment]
Good to know, @ryuash01. Can you share the errors you received?

**Comment by AndrewCraswell** [comment]
I can confirm what @ryuash01 is saying, and our setups are the same. I was already on `6.4.3`, but since `6.4.4` was a minor change, it automatically was installed, and we started seeing 100% test failure. Downgrading and locking to version `6.4.3` fixed the issue.

One tip to others who try to downgrade, I did have to run `jest` with the `--no-cache` option after downgrading in order to clear the cache and start seeing passing tests. Tripped me up for a second.

**Comment by AndrewCraswell** [comment]
We're using `create-react-app` version `4.0.3`. So whichever version they install I think

**Comment by dilanx** [comment]
I just checked out the [v4.0.3 tag of `create-react-app`](https://github.com/facebook/create-react-app/releases/tag/v4.0.3) and looked at the [`package.json` of `react-scripts`](https://github.com/facebook/create-react-app/blob/f92c37a6e6636ed1650ffd976c6881f59b1be803/packages/react-scripts/package.json) at that point in the history. It looks like `babel-jest` 26 is used here. Sorry for missing this!

I think the best option here is to release a new patch that resets the changes made in 6.4.4, which I can try to do right now.

As for @klougod and others that have this issue, I suspect that this issue occurred because you have an ejected CRA project and have updated `babel-jest` independently of CRA, so I think this issue might not relate to craco specifically and rather the version of `babel-jest` you're using. Craco is only really intended to be used with CRA projects that are **not** ejected anyways. Please let me know if there are any other issues.

**Comment by dilanx** [comment]
@klougod okay that makes sense. That's my fault for forgetting to ask which version of `react-scripts` you were using. I just released craco [v6.4.5](https://github.com/dilanx/craco/releases/tag/v6.4.5) which should be the same as v6.4.3 (if you unpublish a version with npm you're unable to use the version tag ever again so it's usually better to always increase the version). Please let me know if there are any other issues.
